client/app/scripts/superdesk-workspace/content/content.js

```diff
@@ -146,6 +148,16 @@
                     });
                 });
 
+                scope.$watch(function() {
+                    return scope.reloadTemplatesList;
```

Comment:
I tried that way first, but it can't be done that way, because there are lots of promise calls, so list would be refreshed before new template is added to list. That's why I did it this way..